product-service/src/post-product.js

```diff
@@ -0,0 +1,36 @@
+import { postProductDB } from './db';
+import allowedOrigins from './constants/alloweb-origins';
+
+export const postProduct = async (event) => {
+  console.log('Lambda invocation with event: ', event);
+  const responce = {
+    statusCode: 200
+  };
+
+  try {
+    const origin = event.headers.origin;
+    if (allowedOrigins.includes(origin)) {
+      responce.headers = {
+        'Access-Control-Allow-Origin': origin,
+        'Access-Control-Allow-Credentials': true,
+      }
+    }
+    const requestBody = JSON.parse(event.body);
+    const result = await postProductDB(requestBody);
+    if (result) {
+      responce.body = JSON.stringify(
+        result,
+        null,
+        2
+      );
+    } else {
+      responce.statusCode = 400;
+      responce.body = 'Can not create product';
```

Comment:
I would better return validation error (to understand what was wrong)

Reply:
The validation error in returned by JOI and returned with 400 status and message from catch